packages/vx-zoom/src/Zoom.tsx

```diff
@@ -10,36 +9,113 @@ import {
+  transformMatrix?: TransformMatrix;
+  /**
+   * By default passive is `false`. This will wrap <Zoom> children in a <div> and add an active wheel
+   * event listener (handleWheel). `handleWheel()` will call `event.preventDefault()` before other
+   * execution. This prevents an outer parent from scrolling when the mouse wheel is used to zoom.
+   *
+   * When passive is `true` it is required to add `<MyComponent onWheel={zoom.handleWheel} />` to handle
+   * wheel events. **Note:** By default you do not need to add `<MyComponent onWheel={zoom.handleWheel} />`.
+   * This is only necessary when `<Zoom passive={true} />`.
+   */
+  passive?: boolean;
+  /** style object to apply to zoom div container. */
+  style?: React.CSSProperties;
+  /** className to apply to zoom div container. */
+  className?: string;
+  children: (zoom: ProvidedZoom & ZoomState) => React.ReactNode;
+};
 
-    this.state = {
-      initialTransformMatrix: props.transformMatrix,
-      transformMatrix: props.transformMatrix,
-      isDragging: false,
-    };
+type ZoomState = {
+  initialTransformMatrix: TransformMatrix;
+  transformMatrix: TransformMatrix;
+  isDragging: boolean;
+};
 
-    this.toString = this.toString.bind(this);
-    this.clear = this.clear.bind(this);
-    this.center = this.center.bind(this);
-    this.handleWheel = this.handleWheel.bind(this);
-    this.dragStart = this.dragStart.bind(this);
-    this.dragMove = this.dragMove.bind(this);
-    this.dragEnd = this.dragEnd.bind(this);
-    this.reset = this.reset.bind(this);
-    this.constrain = props.constrain ? props.constrain.bind(this) : this.constrain.bind(this);
-    this.scale = this.scale.bind(this);
-    this.translate = this.translate.bind(this);
-    this.translateTo = this.translateTo.bind(this);
-    this.setTranslate = this.setTranslate.bind(this);
-    this.setTransformMatrix = this.setTransformMatrix.bind(this);
-    this.invert = this.invert.bind(this);
-    this.applyToPoint = this.applyToPoint.bind(this);
-    this.applyInverseToPoint = this.applyInverseToPoint.bind(this);
-    this.toStringInvert = this.toStringInvert.bind(this);
-  }
+class Zoom extends React.Component<ZoomProps, ZoomState> {
+  static defaultProps = {
+    passive: false,
+    scaleXMin: 0,
+    scaleXMax: Infinity,
+    scaleYMin: 0,
+    scaleYMax: Infinity,
+    transformMatrix: {
+      scaleX: 1,
+      scaleY: 1,
+      translateX: 0,
+      translateY: 0,
+      skewX: 0,
+      skewY: 0,
+    },
+    wheelDelta: (event: React.WheelEvent | WheelEvent) => {
+      return -event.deltaY > 0 ? { scaleX: 1.1, scaleY: 1.1 } : { scaleX: 0.9, scaleY: 0.9 };
+    },
+    style: undefined,
+    className: undefined,
+  };
+
+  containerRef: HTMLDivElement | null = null;
+
+  startPoint: Point | undefined = undefined;
+
+  startTranslate: Translate | undefined = undefined;
+
+  state = {
+    initialTransformMatrix: this.props.transformMatrix!,
+    transformMatrix: this.props.transformMatrix!,
```

Comment:
it can be just 
`
initialTransformMatrix: this.props.transformMatrix,
transformMatrix: this.props.transformMatrix,
`

Reply:
this actually needs the `!` because we don't have any `defaultProps` helpers so it doesn't know they are defined.